### PR TITLE
[exporter/signalfxexporter] Drop datapoints with more than 36 dimensions

### PR DESCRIPTION
**Pull request by dloucasfx**
Signed-off-by: Dani Louca <dlouca@splunk.com>

**Description:** <Describe what has changed.>
The signalfx backend limits the number of allowed dimension to 36 https://docs.splunk.com/Observability/metrics-and-metadata/metric-names.html#dimension-name-and-value-standards
If a datapoint has more than 36 dimensions, signalfx ingest drops the datapoint as invalid without any feedback to the user.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Unit test was added
**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/signalfxexporter/internal/translation/converter.go [hunk 1]
```diff
@@ -244,6 +247,18 @@ func (dpv *datapointValidator) isValidMetricName(name string) bool {
 	return true
 }
 
+func (dpv *datapointValidator) isValidNumberOfDimension(dp *sfxpb.DataPoint) bool {
+	if len(dp.Dimensions) > maxNumberOfDimensions {
+		dpv.logger.Warn("dropping datapoint",
```
Comment (rmfitzpatrick) [inline exporter/signalfxexporter/internal/translation/converter.go:252]: For high throughput metrics this will flood logs and I think debug would be preferable.

File: .chloggen/signalfx-exporter.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
```
Comment (rmfitzpatrick) [inline .chloggen/signalfx-exporter.yaml:2]: nit, I'd think this is an enhancement

File: exporter/signalfxexporter/internal/translation/converter.go (hunk 1 above)
Comment (dloucasfx) [reply exporter/signalfxexporter/internal/translation/converter.go:252]: true, I was just following the current standard in that file.
While I am at it, shall I change the others from warning to debug?

File: exporter/signalfxexporter/internal/translation/converter_test.go [hunk 3]
```diff
@@ -726,6 +726,32 @@ func TestDimensionKeyCharsWithPeriod(t *testing.T) {
...
+	mdInvalid := pmetric.NewMetrics()
+	mInvalid := mdInvalid.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+	m.SetName("invalid")
+	dpInvalid := mInvalid.SetEmptyGauge().DataPoints().AppendEmpty()
+	dp.SetIntValue(123)
+	for i := 0; i < 37; i++ {
+		dpInvalid.Attributes().PutString(fmt.Sprint("dim_key_", i), fmt.Sprint("dim_val_", i))
+	}
+
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (dloucasfx) [reply exporter/signalfxexporter/internal/translation/converter_test.go:751]: good catch , thanks!

File: .chloggen/signalfx-exporter.yaml (hunk 2 above)
Comment (dloucasfx) [reply .chloggen/signalfx-exporter.yaml:2]: 👍

File: exporter/signalfxexporter/internal/translation/converter.go [hunk 4]
```diff
@@ -234,7 +237,7 @@ func (dpv *datapointValidator) sanitizeDimensions(dimensions []*sfxpb.Dimension)
 
 func (dpv *datapointValidator) isValidMetricName(name string) bool {
 	if len(name) > maxMetricNameLength {
-		dpv.logger.Warn("dropping datapoint",
+		dpv.logger.Debug("dropping datapoint",
```
Comment (dmitryax) [inline exporter/signalfxexporter/internal/translation/converter.go:240]: These changes deserve a changelog entry. Can you please submit another PR with them?

Comment (dloucasfx) [reply exporter/signalfxexporter/internal/translation/converter.go:240]: @dmitryax I update the current changelog and added this, let me know if this does it

```
subtext: Additionally, the metric and dimension validation failures are now logged in Debug instead of Warn
```

File: exporter/signalfxexporter/internal/translation/converter_test.go [hunk 5]
```diff
@@ -726,6 +728,57 @@ func TestDimensionKeyCharsWithPeriod(t *testing.T) {
...
+		dpSFX.Dimensions = append(dpSFX.Dimensions, &sfxpb.Dimension{
+			Key:   fmt.Sprint("dim_key_", i),
+			Value: fmt.Sprint("dim_val_", i),
+		})
+	}
+	assert.EqualValues(t, 0, len(c.MetricsToSignalFxV2(mdInvalid)))
+	// Check log results:
+	// 1- Only a single log message
+	// 2- Verify that the one log message is equal to "dropping datapoint"
+	// 3- Verify that the fields match
```
Comment (rmfitzpatrick) [inline exporter/signalfxexporter/internal/translation/converter_test.go:772]: nit, imo these comments are unnecessary since the assertions are pretty clear

**Review by rmfitzpatrick — APPROVED** [review]

File: exporter/signalfxexporter/internal/translation/converter_test.go (hunk 5 above)
Comment (dloucasfx) [reply exporter/signalfxexporter/internal/translation/converter_test.go]: removed

**Review by dmitryax — APPROVED** [review]
